Frameworks/ImageIO/CGImageSource.mm

```diff
@@ -112,6 +115,890 @@ - (CFStringRef)getImageType {
+                           forKey:(id)kCGImagePropertyExifMakerNote];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/ifd/exif/{ushort=37383}", &propertyValue)) && propertyValue.vt == VT_UI2) {
+        [exifProperties setObject:[NSNumber numberWithInt:propertyValue.uiVal] forKey:(id)kCGImagePropertyExifMeteringMode];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/ifd/exif/{ushort=41990}", &propertyValue)) && propertyValue.vt == VT_UI2) {
+        [exifProperties setObject:[NSNumber numberWithInt:propertyValue.uiVal] forKey:(id)kCGImagePropertyExifSceneCaptureType];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/ifd/exif/{ushort=37377}", &propertyValue)) && propertyValue.vt == VT_I8) {
+        [exifProperties setObject:[NSNumber numberWithDouble:(double)propertyValue.hVal.LowPart/propertyValue.hVal.HighPart] 
+                           forKey:(id)kCGImagePropertyExifShutterSpeedValue];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/ifd/exif/{ushort=37510}", &propertyValue)) && propertyValue.vt == VT_LPWSTR) {
+        [exifProperties setObject:[[NSString alloc] initWithBytes:propertyValue.pwszVal 
+                                                           length:wcslen(propertyValue.pwszVal)*sizeof(wchar_t)
+                                                         encoding:NSUnicodeStringEncoding] 
+                           forKey:(id)kCGImagePropertyExifUserComment];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/ifd/exif/{ushort=36864}", &propertyValue)) && propertyValue.vt == VT_BLOB) {
+        NSData* exifVersion = [NSData dataWithBytesNoCopy:propertyValue.blob.pBlobData length:propertyValue.blob.cbSize freeWhenDone:YES];
+        NSMutableArray* exifVersionArray = [NSMutableArray array];
+        char* exifVersionCharacters = (char*)[exifVersion bytes];
+        [exifVersionArray addObject:[NSNumber numberWithInt:exifVersionCharacters[1] - '0']];
+        [exifVersionArray addObject:[NSNumber numberWithInt:exifVersionCharacters[2] - '0']];
+        [exifVersionArray addObject:[NSNumber numberWithInt:exifVersionCharacters[3] - '0']];
+        [exifProperties setObject:exifVersionArray forKey:(id)kCGImagePropertyExifVersion];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/ifd/exif/{ushort=41987}", &propertyValue)) && propertyValue.vt == VT_UI2) {
+        [exifProperties setObject:[NSNumber numberWithInt:propertyValue.uiVal] forKey:(id)kCGImagePropertyExifWhiteBalance];
+    }
+
+    // Add the Exif dictionary to the properties if there are any Exif properties
+    if ([[exifProperties allKeys] count] != 0) {
```

Comment:
same question applies in several spots...

---
In reply to: [60454530](https://github.com/Microsoft/WinObjC/pull/445#discussion_r60454530) [](ancestors = 60454530)